frontend/src/app/projects/models/create-coworking-method/create-coworking-method.component.ts

```diff
@@ -13,58 +20,139 @@ import {
 } from 'src/app/services/git/git.service';
 import { ModelService } from 'src/app/services/model/model.service';
 import { Source, SourceService } from 'src/app/services/source/source.service';
-import { filter, switchMap, map, combineLatest } from 'rxjs';
+import {
+  GitSetting,
+  GitSettingsService,
+} from 'src/app/services/settings/git-settings.service';
+import {
+  absoluteOrRelativeSafetyValidators,
+  absoluteUrlSafetyValidator,
+  checkUrlForInvalidSequences,
+  hasAbsoluteUrlPrefix,
+  hasRelativePathPrefix,
+} from 'src/app/helpers/validators/url-validator';
 
 @Component({
   selector: 'app-create-coworking-method',
   templateUrl: './create-coworking-method.component.html',
   styleUrls: ['./create-coworking-method.component.css'],
 })
 export class CreateCoworkingMethodComponent implements OnInit {
-  @Output() create = new EventEmitter<boolean>();
+  @Output() create = new EventEmitter<{ created: boolean }>();
+
+  public availableGitSettings: Array<GitSetting> = [];
+  public selectedGitSetting: GitSetting | undefined = undefined;
+
+  private availableRevisions: Instance | undefined = {
+    branches: [],
+    tags: [],
+  };
+  public filteredRevisions: Instance = { branches: [], tags: [] };
+
+  public resultUrl: string = '';
 
   public form = new FormGroup({
+    urlGroup: new FormGroup({
+      baseUrl: new FormControl<GitSetting | undefined>(undefined),
+      inputUrl: new FormControl('', absoluteUrlSafetyValidator()),
+    }),
     credentials: new FormGroup({
-      path: new FormControl('', Validators.required),
       username: new FormControl(''),
       password: new FormControl(''),
     }),
-    revision: new FormControl('', Validators.required),
+    revision: new FormControl('', [
+      Validators.required,
+      this.validRevisionValidator(),
+    ]),
     entrypoint: new FormControl('/'),
   });
 
-  public filteredRevisions: Instance = { branches: [], tags: [] };
-
   constructor(
     public projectService: ProjectService,
     public modelService: ModelService,
+    public gitSettingsService: GitSettingsService,
     private gitService: GitService,
     private sourceService: SourceService
   ) {}
 
   ngOnInit(): void {
-    this.form.controls.revision.valueChanges.subscribe((value) => {
-      if (!this.gitService.instance) {
-        this.filteredRevisions = { branches: [], tags: [] };
-      } else {
-        this.filteredRevisions = {
-          branches: this.gitService.instance.branches.filter((branch) =>
-            branch.startsWith(value as string)
-          ),
-          tags: this.gitService.instance.tags.filter((tag) =>
-            tag.startsWith(value as string)
-          ),
-        };
+    this.gitService.instance.subscribe((revisions) => {
+      this.availableRevisions = revisions;
+      this.form.controls.revision.updateValueAndValidity({
+        onlySelf: false,
+        emitEvent: true,
+      });
```

Comment:
```suggestion
      this.form.controls.revision.updateValueAndValidity();
```

Should be sufficient.